Source/WebKit/Shared/Cocoa/SandboxInitialiationParametersCocoa.mm

```diff
@@ -60,44 +54,48 @@
     appendPathInternal(name, path);
 }
 
-void SandboxInitializationParameters::addPathParameter(const char* name, NSString *path)
+void SandboxInitializationParameters::addPathParameter(ASCIILiteral name, NSString *path)
 {
     appendPathInternal(name, [path length] ? [(NSString *)path fileSystemRepresentation] : "");
 }
 
-void SandboxInitializationParameters::addPathParameter(const char* name, const char* path)
+void SandboxInitializationParameters::addPathParameter(ASCIILiteral name, const char* path)
 {
     appendPathInternal(name, path);
 }
 
-void SandboxInitializationParameters::addParameter(const char* name, const char* value)
+void SandboxInitializationParameters::addParameter(ASCIILiteral name, CString&& value)
 {
-    m_namedParameters.append(name);
-    m_namedParameters.append(fastStrDup(value));
+    m_parameterNames.append(name);
+    m_parameterValues.append(WTFMove(value));
 }
 
-const char* const* SandboxInitializationParameters::namedParameterArray() const
+Vector<const char*> SandboxInitializationParameters::namedParameterVector() const
 {
-    if (!(m_namedParameters.size() % 2))
-        m_namedParameters.append(static_cast<const char*>(0));
-
-    return m_namedParameters.data();
+    Vector<const char*> result;
+    result.reserveInitialCapacity(m_parameterNames.size() * 2 + 1);
+    ASSERT(m_parameterNames.size() == m_parameterValues.size());
+    for (size_t i = 0; i < m_parameterNames.size(); ++i) {
+        result.append(m_parameterNames[i]);
+        result.append(m_parameterValues[i].data());
+    }
+    result.append(nullptr);
```

Comment:
Is this vector item required?

Reply:
Answering my own question, I think it is, since the sandbox function sandbox_init_with_parameters does not provide a size parameter.